src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -28,99 +43,640 @@ const ActorInit En_Bigokuta_InitVars = {
+        }
+
+        this->playerHoldPos.x = (Math_SinS(this->actor.shape.rot.y) * 66.0f) + this->actor.world.pos.x;
+        this->playerHoldPos.y = (this->actor.home.pos.y - 49.5f) + 42.899998f;
+        this->playerHoldPos.z = (Math_CosS(this->actor.shape.rot.y) * 66.0f) + this->actor.world.pos.z;
+
+        Math_Vec3f_Copy(&player->actor.world.pos, &this->playerHoldPos);
+        Math_Vec3f_Copy(&this->playerPos, &player->actor.world.pos);
+        player->actor.shape.rot.y += BINANG_SUB(this->actor.shape.rot.y, rotY);
+    } else {
+        Math_Vec3f_Copy(&this->playerPos, &player->actor.world.pos);
+        player->actor.velocity.y = 0.0f;
+    }
+}
+
+void EnBigokuta_SetupHoldPlayer(EnBigokuta* this) {
+    this->timer = 12;
+    this->actionFunc = EnBigokuta_HoldPlayer;
+}
+
+void EnBigokuta_HoldPlayer(EnBigokuta* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->timer--;
+    if (this->timer >= 0) {
+        EnBigokuta_UpdateOrSetupCam(this, globalCtx);
+        Math_Vec3f_Copy(&player->actor.world.pos, &this->playerHoldPos);
+
+        if (this->timer == 0) {
+            EnBigokuta_ShootPlayer(this, globalCtx);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_REVERSE);
+        }
+    } else if (this->timer == -24) {
+        EnBigokuta_SetupIdleAboveWater(this);
+    }
+}
+
+void EnBigokuta_SetupDeathCutscene(EnBigokuta* this) {
+    ActorCutscene_SetIntentToPlay(this->cutscene);
+    this->timer = 0;
+    this->bodyCollider.base.acFlags &= ~AC_ON;
+    this->actionFunc = EnBigokuta_PlayDeathCutscene;
+}
+
+void EnBigokuta_PlayDeathCutscene(EnBigokuta* this, GlobalContext* globalCtx) {
+    Player* player;
+
+    this->actor.colorFilterTimer = Animation_GetLastFrame(&gBigOctoDeathAnim);
+
+    if (this->timer != 0) {
+        this->timer--;
+        if (this->timer == 0) {
+            this->drawEffect = 0;
+            this->drawEffectAlpha = 0.0f;
+            func_800BF7CC(globalCtx, &this->actor, this->limbPos, ARRAY_COUNT(this->limbPos), 2, 0.5f, 0.35f);
+            EnBigokuta_SetupDeathEffects(this);
+        }
+    } else if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+        ActorCutscene_Start(this->cutscene, &this->actor);
+
+        if (!(gSaveContext.eventInf[4] & 2) && !(gSaveContext.eventInf[3] & 0x20)) {
+            func_800B724C(globalCtx, &this->actor, 7);
+        } else {
+            player = GET_PLAYER(globalCtx);
+            player->stateFlags1 |= 0x20;
+        }
+
+        if (this->drawEffect == BIGOCTO_DRAWEFFECT_ICE) {
+            this->timer = 3;
+        } else {
+            EnBigokuta_SetupDeathEffects(this);
+        }
+    } else {
+        ActorCutscene_SetIntentToPlay(this->cutscene);
+    }
+}
+
+void EnBigokuta_SetupDeathEffects(EnBigokuta* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gBigOctoDeathAnim, -5.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_DEAD2);
+    this->actor.flags &= -2;
```

Comment:
```suggestion
    this->actor.flags &= ~1;
```
I believe this should work?